python/hail/docs/tutorials/03-expressions.ipynb

```diff
@@ -142,6 +241,91 @@
     "three.show()"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Missing data\n",
+    "\n",
+    "All expressions in Hail can represent missing data. Hail has a [collection of primitive operations](https://hail.is/docs/devel/functions/core.html) for dealing with missingness. \n",
+    "\n",
+    "The [null](https://hail.is/docs/devel/functions/core.html?highlight=null#hail.expr.functions.null) method can be used to create a missing expression of a specific type, such as a missing string:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "missing_string = hl.null(hl.tstr)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Expressions handle missingness in the following ways:\n",
+    "\n",
+    "* a missing value plus another value is always missing\n",
+    "* a conditional statement with a missing predicate is missing\n",
+    "* when aggregating a sum of values, the missing values are ignored\n",
+    "\n",
+    "This is different from Python's treatment of missingness, where `None + 5` would produce an error. In Hail, `hl.null(hl.tint32) + 5` produces a missing result, not an error. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hl.is_missing(hl.null(hl.tint32)+5).value"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here are a few more examples to illustrate how missingness is treated in Hail:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hl.is_defined(hl.str(\"not missing\")).value"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hl.sum(hl.array([1, 2, hl.null(hl.tint32)])).value"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hl.or_missing(True, hl.int32(5)).value"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(hl.or_missing(False, hl.int32(5)).value)"
```

Comment:
Does this need the print statement?

Reply:
unfortunately, it won't print `None` unless the print is there.